__tests__/base.spec.ts

```diff
@@ -1,5 +1,33 @@
-describe("Testing tests", () => {
-  test("Basic test", (done) => {
-    done();
-  });
+var assert = require('assert');
+var { ComplimentGenerator, InsultGenerator, JokeGenerator } = require("../src/generators")
+
+describe("Generator tests", () => {
+  it('Compliment generator', async function() {
+    const generator = new ComplimentGenerator()
+    
+    const data = await generator.generate()
+
+    // Check if url source is initialized
+    assert.ok(generator.source.length != 0);
+    // Check if data is retrived from source
+    assert.ok(data.length != 0)
```

Comment:
For all tests, also assert the existence of the corresponding object (data.compliment, data.setup, etc.)

Reply:
done